### PR TITLE
Shared tests

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: https://github.com/devtools-html/debugger.html/issues/5100

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

* Adding tests for various components in the `shared/` folder
  * shared/Button components
  * sharedAccordion
  * shared/Badge
  * shared/BracketArrow
  * shared/Dropdown
  * shared/SearchInput
  * shared/Popover

### CODE REVIEW

File: src/components/shared/Accordion.js [hunk 1]
```diff
@@ -58,20 +55,17 @@ class Accordion extends Component<Props, State> {
     return (
       <div className={item.className} key={i}>
         <div className="_header" onClick={() => this.handleHeaderClick(i)}>
-          <Svg name="arrow" className={opened ? "expanded" : ""} />
+          <Svg name="arrow" className={opened === true ? "expanded" : ""} />
```
Comment (jasonLaster) [inline src/components/shared/Accordion.js]: `opened === true` this feels unnecessary if we know it's a bool

File: src/components/shared/Button/PaneToggle.js [hunk 2]
```diff
@@ -12,17 +12,20 @@ import "./PaneToggle.css";
...
-
-    return (
-      horizontal !== nextProps.horizontal || collapsed !== nextProps.collapsed
-    );
+    const { collapsed, horizontal, position } = this.props;
+    const diffHorizontal = horizontal !== nextProps.horizontal;
+    const diffCollapsed = collapsed !== nextProps.collapsed;
+    const diffPosition = position !== nextProps.position;
+    return diffHorizontal || diffCollapsed || diffPosition;
   }
```
Comment (jasonLaster) [inline src/components/shared/Button/PaneToggle.js]: can this be a `PureComponent`?

File: src/components/shared/Popover.js [hunk 3]
```diff
@@ -54,49 +63,59 @@ class Popover extends Component {
...
-      orientation == "down"
-        ? targetRect.bottom
-        : targetRect.top - popoverRect.height;
-
-    const targetMid = targetRect.left - popoverLeft + targetRect.width / 2 - 8;
-
-    return { left: popoverLeft, top, orientation, targetMid };
+    if (this.$popover && this.props.editorRef) {
+      const popover = this.$popover;
+      const editor = this.props.editorRef;
```
Comment (jasonLaster) [inline src/components/shared/Popover.js]: nice

File: src/components/shared/tests/Button.spec.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    ).toMatchSnapshot();
+  });
+});
+
+describe("PaneToggleButton", () => {
+  const spy = jest.fn();
+  const wrapper = shallow(
+    <PaneToggleButton collapsed={false} handleClick={spy} position="start" />
+  );
+  test("renders start", () => expect(wrapper).toMatchSnapshot());
```
Comment (jasonLaster) [inline src/components/shared/tests/Button.spec.js]: why test and not `it`

File: src/components/shared/Button/PaneToggle.js (hunk 2 above)
Comment (wldcordeiro) [reply src/components/shared/Button/PaneToggle.js]: Yeah good point the prop types are simple data types. 👍

File: src/components/shared/tests/Button.spec.js (hunk 4 above)
Comment (wldcordeiro) [reply src/components/shared/tests/Button.spec.js]: I heard some rumblings of `it` being deprecated by Jest and they don't directly mention it as part of their global API. http://facebook.github.io/jest/docs/en/api.html

Comment (jasonLaster) [reply src/components/shared/tests/Button.spec.js]: Got it. How about switching to `it` here to be consistent and open an issue to discuss moving to `test`...

**Pull request merged**